cli/cliutils/cliutils.go

```diff
@@ -1555,28 +1556,60 @@ func VerifySigningKeyInput(keyFile string, isPublic bool) string {
 
 // get default keys if needed and verify them.
 // this function is used by `hzn exhcange pattern/service publish
-func GetSigningKeys(privKeyFilePath, pubKeyFilePath string) (string, string) {
+func GetSigningKeys(privKeyFilePath, pubKeyFilePath string) (string, []byte, string) {
 
-	// if the -k is specified but -K is not specified, then do not get public key default.
-	// the public key will not be stored with the resource.
-	defaultPublicKey := true
-	if privKeyFilePath != "" && pubKeyFilePath == "" {
-		defaultPublicKey = false
+	var err error
+	
+	// if -K is not specified, then the public key will be generated from the private key.
+	publicKeyGiven := true
+	if pubKeyFilePath == "" {
+		publicKeyGiven = false
 	}
 
-	// get default private key
+	// Get default private key if -k not specified
+	var privKey *rsa.PrivateKey
 	privKeyFilePath_tmp := WithDefaultEnvVar(&privKeyFilePath, "HZN_PRIVATE_KEY_FILE")
 	privKeyFilePath = VerifySigningKeyInput(*privKeyFilePath_tmp, false)
-
-	if privKeyFilePath != "" {
-		verifyPrivateKeyFormat(privKeyFilePath)
+	if privKey, err = sign.ReadPrivateKey(privKeyFilePath); err != nil {
```

Comment:
I originally stopped using it because it did not return the `*rsa.PrivateKey` object I needed to extract the public key. Now, I refactored the function to return the `*rsa.PrivateKey` object because this is the only place the function is called anyway, so it won't interfere with any other code.